sssd_test_framework/utils/authentication.py

```diff
@@ -320,22 +320,30 @@ def passkey(self, username: str, *, pin: str | int, device: str, ioctl: str, scr
         :return: True if authentication was successful, False otherwise.
         :rtype: bool
         """
-        self.fs.backup("/etc/sysconfig/sssd")
+        self.fs.backup("/usr/libexec/sssd/passkey_child")
+        self.fs.copy("/usr/libexec/sssd/passkey_child", "/usr/libexec/sssd/passkey_child.orig")
+        self.host.ssh.run("rm -f /usr/libexec/sssd/passkey_child", raise_on_error=False)
+
         device_path = self.fs.upload_to_tmp(device, mode="a=r")
         ioctl_path = self.fs.upload_to_tmp(ioctl, mode="a=r")
         script_path = self.fs.upload_to_tmp(script, mode="a=r")
 
         run_su = self.fs.mktmp(
             rf"""
-            #!/bin/bash
-            set -ex
-            env | grep ^UMOCKDEV_ > /etc/sysconfig/sssd
-            printf "LD_PRELOAD=$LD_PRELOAD" >> /etc/sysconfig/sssd
-            systemctl restart sssd
-            chmod -R a+rwx $UMOCKDEV_DIR
-
-            su --shell /bin/sh nobody -c "su - '{username}'"
-            """,
+                #!/bin/bash
+                set -ex
+                echo '#!/bin/bash' > /usr/libexec/sssd/passkey_child
+                echo -n 'export ' >> /usr/libexec/sssd/passkey_child
+                env | grep ^UMOCKDEV_ >> /usr/libexec/sssd/passkey_child
+                echo -n 'export ' >> /usr/libexec/sssd/passkey_child
+                printf "LD_PRELOAD=$LD_PRELOAD\n" >> /usr/libexec/sssd/passkey_child
+                echo 'exec /usr/libexec/sssd/passkey_child.orig $@' >> /usr/libexec/sssd/passkey_child
+                chmod 755 /usr/libexec/sssd/passkey_child
```

Comment:
You can move this in `self.fs.write(...)` and remove `self.fs.backup("/usr/libexec/sssd/passkey_child")`

Reply:
@pbrezina we might be able to predict what some of the UMOCKDEV_* vars will be based on the device info but, I'm not sure we can predict UMOCKDEV_DIR.   That will differ everytime umockdev-run is executed.

Without creating the passkey_child every time with the var for that execution like above, how can we make sure the passkey_child contains the correct value for UMOCKDEV_DIR?    I'm guessing since it's SSSD that executes passkey_child, that it inherits it's env from there right?

Reply:
content_run_su = """
                #!/bin/bash
                export LD_PRELOAD=/opt/random.so:libumockdev-preload.so.0
                exec /usr/libexec/sssd/passkey_child.orig $@ \n
                """
        self.fs.write("/usr/libexec/sssd/passkey_child", content_run_su, mode="0755")

        run_su = self.fs.mktmp(
            rf"""
                #!/bin/bash
                set -ex
                echo "" >> /usr/libexec/sssd/passkey_child
                echo -n 'export ' >> /usr/libexec/sssd/passkey_child
                env | grep ^UMOCKDEV_ >> /usr/libexec/sssd/passkey_child
                #systemctl restart sssd
                chmod -R a+rwx $UMOCKDEV_DIR
                su --shell /bin/sh nobody -c "su - '{username}'"
                """,
            mode="a=rx",
        )

hello @pbrezina, I try this way, but its not working, as half part with `fs.write` and other with `self.fs.mktmp` getting auth failure.

Reply:
hmm, ok. Then disregard this comment.